### DISCUSSION TITLE
New ports

### DISCUSSION THREAD

**Discussion by atsushieno** [atsushieno/android-audio-plugin-framework#98] [Show and tell]
New ports would be posted here.

There was a former thread in aap-juce https://github.com/atsushieno/aap-juce/discussions/31, but it makes much more sense if we have it in this repository instead. (update: aap-juce discussions is now disabled so it's 404. I was the only poster there.)

**Comment by atsushieno** [comment]
String-Machine is ported as a new aap-lv2 plugin: https://github.com/atsushieno/aap-lv2-string-machine

It is primarily to see if DPF plugin port works or not. (It's working quite well.)

![image](https://user-images.githubusercontent.com/53929/154757107-1f6f72e1-3e50-4e7a-bdfe-ad1906f28eb1.png)

**Comment by atsushieno** [comment]
Now there is a Wiki page that lists all known plugins and hosts. https://github.com/atsushieno/android-audio-plugin-framework/wiki/List-of-AAP-plugins-and-hosts

**Comment by atsushieno** [comment]
I didn't have a lot of successful plugin ports last year, whereas I actually tried some...

Today I got OS-251 working. It is a react-juce plugin, which means now we have a real-world example of react-juce plugin in the AAP world. It is available as [aap-juce-os251](https://github.com/atsushieno/aap-juce-os251).

![image](https://user-images.githubusercontent.com/53929/212149240-c9361c9d-e55f-4583-9f13-be529f3a6f59.png) ![image](https://user-images.githubusercontent.com/53929/212149272-2cb796ba-9527-4eef-b787-c3b8df025590.png) ![image](https://user-images.githubusercontent.com/53929/212149297-f1a181a0-ca88-47b2-9f01-71c05b1c1704.png)

**Comment by atsushieno** [comment]
I kinda got BYOD working (if I can say it's working while MainActivity fails to launch, loading presets takes minutes, and applying effect also takes minute-ish). Code is up at https://github.com/atsushieno/aap-juce-byod

![image](https://user-images.githubusercontent.com/53929/212462606-2494cc99-55fb-42ef-9187-edba23f011be.png)

**Reply by atsushieno** [reply]
I made it better, with fix for RTNeural (used by BYOD). Now it applies effect in good performance just as other effect plugins do...!

**Comment by atsushieno** [comment]
It is not really a "new" port, but last week I finally got my AAP integrated fork of helio-workstation working to some extent: https://github.com/atsushieno/aap-juce-helio

Formerly those AAP instruments were never connectable to audio outputs. It was problematic at both AAP and helio sides, our `juceaap::AndroidAudioPluginInstance` were constructed without `juce::BusesProperties` which seems to be "mandatory" for helio (most likely for no reason). aap-juce is modernized to work with it.

**Comment by atsushieno** [comment]
I created a new aap-juce port: https://github.com/atsushieno/aap-juce-peak-eater

It is rather for recording my porting workflow (linked from README).

![image](https://user-images.githubusercontent.com/53929/216549592-6a9eac75-47e7-413e-b869-cea66a39e967.png)

**Comment by atsushieno** [comment]
I'm not sure if it is ever usable, but done with another porting experiment in 1h ~ish. https://github.com/atsushieno/aap-juce-seele

![image](https://user-images.githubusercontent.com/53929/217461162-5cfc41d7-2016-450a-9e0c-3e0b91d22e54.png)

**Comment by atsushieno** [comment]
Another month, another plugin port: [aap-juce-xenos](https://github.com/atsushieno/aap-juce-xenos)

![image](https://user-images.githubusercontent.com/53929/229281108-d4386532-7b2b-465c-871f-cdd1cd4cecc1.png)

**Comment by atsushieno** [comment]
Recently I created a JUCE CMake port of ADLplug (which may sound weird as there is already CMakeLists.txt, but it is a customized build script that is independent of JUCE CMake setup) and labelled as ADLplug-AE, as it brings in incompatibility on LV2 version (ports are set up in JUCE LV2 manner).

aap-juce-adlplug has been unable to build with the latest AGP toolchain somehow, so it is replaced with this new port (it does not really change except for the plugin name, in terms of AAP): https://github.com/atsushieno/aap-juce-adlplug-ae/

**Comment by atsushieno** [comment]
And, magenta/ddsp-vst is now ported to AAP: https://github.com/atsushieno/aap-juce-ddsp

![image](https://github.com/atsushieno/aap-core/assets/53929/de319914-6431-42a2-aff9-f3c417ee2939)

**Comment by atsushieno** [comment]
After various patching around, AIDA-X builds and kind of runs as AAP: https://github.com/atsushieno/aap-lv2-aida-x

It still seems heavyweight though. DPF DGL does not build on Android, so no GUI either.

<img width="428" alt="image" src="https://github.com/atsushieno/aap-core/assets/53929/283da9cd-7cc8-49e0-a266-e3dd9745c107">

**Comment by atsushieno** [comment]
With the recent changes that makes generate-aap-metadata step optional, aap-juce-surge is finally working.

![image](https://github.com/atsushieno/aap-core/assets/53929/f7758587-aff4-4f7f-b409-5f366d2beff7)
![image](https://github.com/atsushieno/aap-core/assets/53929/a0a7368b-4669-451f-bb5a-bffd09b59d73)

**Comment by atsushieno** [comment]
Haven't ported plugins for a while, but it is still doable - [Audible Planets](https://github.com/gregrecco67/AudiblePlanets) ported to AAP: https://github.com/atsushieno/aap-juce-audible-planets

![image](https://github.com/atsushieno/aap-core/assets/53929/b10dfff8-e1b0-4f03-b366-bb74a9c05dd5)


emulator screencast at https://g0v.social/@atsushieno/112269129225298628

**Comment by atsushieno** [comment]
It is a new port that is not either aap-lv2 or aap-juce: [aap-airwindows](https://github.com/atsushieno/aap-airwindows/)

![aap-airwindows sshot](https://github.com/atsushieno/aap-core/assets/53929/859092c9-08f1-4b43-8ff5-a8670d4bad80)

**Comment by atsushieno** [comment]
Vaporizer2 is ported too. Its UI crashes at startup though (DSP seems to work).
https://github.com/atsushieno/aap-juce-vaporizer2

**Comment by atsushieno** [comment]
Apparently I didn't post about new ports here... there are handful of new ports that are posted to our Wiki page now. More importantly, unlike days I posted here, those JUCE ports now come up with working GUI, so you would like to re-evaluate everything - for example, Surge XT was pretty useless without GUI and newly bundled presets. Now it's much nicer.

**Reply by henrebotha** [reply]
It's so cool to see the Surge GUI on my phone lol.

I think your simple AAP host could use a way to zoom/pan the GUI. Right now I can see the Surge GUI, but it's basically unusable because it's so small on my phone screen. Surge has a built-in zoom feature, but if I use that to zoom in, the GUI just gets cropped by the host.

**Reply by atsushieno** [reply]
It is not a reasonable expectation that Surge's desktop GUI can be as usable as on desktop. Taking that in mind, the built-in hosting app in Compose shows the plugin GUI with scrollables, and the resize grip *within* the JUCE gui expands the content.

It's unclear to me how changing zoom within Surge can/should trigger resize events on the hosted window. I would rather like to see code that proves it, beyond some ideas.
